### PR TITLE
Lf 249 log permissions

### PR DESCRIPTION
**Pull request by JimmyRowland**

### CODE REVIEW

File: packages/api/src/controllers/logController.js [hunk 1]
```diff
@@ -79,11 +79,50 @@ class logController extends baseController {
...
+
+  static getLogByFarmId(){
+    return async (req, res) => {
+      try{
+        if(req.params.farm_id){
+          const farm_id = req.params.farm_id
+          const logs = await logServices.getLogByFarm(farm_id);
+          if(logs && !logs.length){
+            //TODO: replace [{}] with []
+            res.json([{}]);
```
Comment (kikemarquez8) [inline packages/api/src/controllers/logController.js:107]: can we do this now? Or is there a front end restriction?

File: packages/api/src/middleware/acl/hasFarmAccess.js [hunk 2]
```diff
@@ -69,6 +73,53 @@ async function fromField(fieldId) {
...
+    return {};
+  }
+  const field_ids = fields?fields.map((field)=> field.field_id):undefined;
+  try{
+    const userFarms = await knex('field').join('userFarm','field.farm_id','userFarm.farm_id').whereIn('field.field_id',field_ids).distinct('field.farm_id');
+    if(userFarms.length!==1) return {};
+    return userFarms[0];
+  }catch (e){
+    return {};
+  }
```
Comment (kikemarquez8) [inline packages/api/src/middleware/acl/hasFarmAccess.js:87]: can you format this code?

File: packages/api/src/middleware/acl/hasFarmAccess.js [hunk 3]
```diff
@@ -69,6 +73,53 @@ async function fromField(fieldId) {
...
+      if(!field.field_id){
+        return {};
+      }
+      fields.push(field.field_id);
+    }
+    if(fields.length===0){
+      return {};
+    }
+  }
+  const userFarmModel = require('../../models/userFarmModel');
```
Comment (kikemarquez8) [inline packages/api/src/middleware/acl/hasFarmAccess.js:107]: Move import to the import section

**Review by kikemarquez8 — CHANGES_REQUESTED** [review]

File: packages/api/src/controllers/logController.js [hunk 4]
```diff
@@ -51,14 +50,15 @@ class logController extends baseController {
   }
 
 
-  static getLog(){
+  static getLog1(){
```
[the remark this answers is not in the document]
Comment (JimmyRowland) [reply packages/api/src/controllers/logController.js]: getLog1 is the old implementation

[the remark this answers is not in the document]
Comment (kikemarquez8) [reply packages/api/src/controllers/logController.js]: can we delete it?

File: packages/api/src/controllers/logController.js (hunk 1 above)
Comment (JimmyRowland) [reply packages/api/src/controllers/logController.js]: fixed

File: packages/api/src/controllers/logController.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (JimmyRowland) [reply packages/api/src/controllers/logController.js]: Deleted

**Review by kikemarquez8 — APPROVED** [review]

**Pull request merged**